proxy/handlers.go

```diff
@@ -501,3 +501,18 @@ func (p *Proxy) GetSignOutURL(authenticateURL, redirectURL *url.URL) *url.URL {
 func extendDeadline(ttl time.Duration) time.Time {
 	return time.Now().Add(ttl).Truncate(time.Second)
 }
+
+// websocketHandlerFunc splits request serving with timeouts depending on the protocol
+func websocketHandlerFunc(baseHandler http.Handler, timeoutHandler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		cn := r.Header.Get("Connection")
+		up := r.Header.Get("Upgrade")
+
+		if r.ProtoMajor == 1 && strings.Contains(cn, "Upgrade") && strings.Contains(up, "websocket") {
```

Comment:
nit: Although the line gets longer, you can probably inline the r.Get()'s which saves the call if r.Proto is not 1.

Reply:
Should we match for exact string for websocket and upgrade vs contains?

Reply:
Good points, I can make both changes tomorrow.